zerver/lib/user_topics.py

```diff
@@ -121,11 +121,9 @@ def exclude_topic_mutes(
         # by not considering topic mutes outside the stream.
         query = query.filter(stream_id=stream_id)
 
-    rows = list(
-        query.values(
-            "recipient_id",
-            "topic_name",
-        )
+    rows = query.values(
+        "recipient_id",
+        "topic_name",
     )
 
     if not rows:
```

Comment:
I'm not sure, but I think removing the `list()` wrapper here might add an extra database query -- I think `if not rows` is going to do a `.exists()` query, and then we'll do a second query when accessing `rows` later.

Reply:
I believe that the result is [cached](https://github.com/django/django/blob/6f80050496780e2f8a65e2a4374dd6375c9fbfa5/django/db/models/query.py#L2070-L2076) once either `__iter__` and `__bool__` is called.